shell/src/main/java/org/apache/kafka/shell/MetadataShell.java

```diff
@@ -119,6 +141,7 @@ private void initializeWithRaftManager() {
     }
 
     private void initializeWithSnapshotFileReader() throws Exception {
+        this.fileLock = takeFileLock(new File(snapshotPath).getParentFile());
```

Comment:
This doesn't look like the correct directory. Please take a look at the `LogManager` for details.

Reply:
Fixed